internal/servers/destinations.go

```diff
@@ -113,9 +113,9 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 		case <-ctx.Done():
 			close(resources)
 			if err := eg.Wait(); err != nil {
-				s.Logger.Error().Err(err).Msg("failed to wait")
+				return status.Errorf(codes.Internal, "Context done: %v and failed to wait for plugin: %v",ctx.Err(),  err)
 			}
-			return ctx.Err()
+			return status.Errorf(codes.Internal, "Context done: %v", ctx.Err())
```

Comment:
See https://github.com/cloudquery/plugin-sdk/pull/486/files for how to convert context errors to gRPC status codes